c7n/resources/asg.py

```diff
@@ -1380,11 +1432,23 @@ def process_asg(self, asg):
         # max tags constraints, otherwise.
         #
         # delete_first = len([t for t in tag_map if not t.startswith('aws:')])
-        client.delete_tags(Tags=[
-            {'ResourceId': asg['AutoScalingGroupName'],
-             'ResourceType': 'auto-scaling-group',
-             'Key': source_tag,
-             'Value': source['Value']}])
+        if isinstance(self.data.get('source'), list):
+            client.delete_tags(
+                Tags=[
+                    {
+                        'ResourceId': asg['AutoScalingGroupName'],
+                        'ResourceType': 'auto-scaling-group',
+                        'Key': s
+                    }
+                    for s in self.data['source']
+                ]
+            )
+        else:
+            client.delete_tags(Tags=[
```

Comment:
this clause goes away since its always a list?

Reply:
there's a few other places in the pr making distinctions on list vs singleton that could be simplified with always a list.